### PR TITLE
Issue #9454: browser-storage-sync implementation for credit card and address storage

### LINKED ISSUE #9454: browser-storage-sync implementation for credit cards and addresses storage
Larger issue: #9452

An a-s backed implementation of https://github.com/mozilla-mobile/android-components/issues/9453

Follow how `PlacesHistoryStorage` and `PlacesBookmarksStorage` are wired up, and `SyncableLoginsStorage` for how to initialize an encrypted store (see `unlock`).

### PR DESCRIPTION
**Pull request by gabrielluong**
Fixes #9454 

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by gabrielluong** [comment]
I am getting the same error locally with uniffi https://firefox-ci-tc.services.mozilla.com/tasks/YHBuXxnUSvOaeI5YYlAgqg/runs/0/logs/https%3A%2F%2Ffirefox-ci-tc.services.mozilla.com%2Fapi%2Fqueue%2Fv1%2Ftask%2FYHBuXxnUSvOaeI5YYlAgqg%2Fruns%2F0%2Fartifacts%2Fpublic%252Flogs%252Flive.log#L941

File: components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/AutofillCreditCardsAddressesStorage.kt [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+/**
+ * An implementation of [CreditCardsAddressesStorage] back by the application-services' `autofill`
+ * library.
+ */
+@SuppressWarnings("TooManyFunctions")
+class AutofillCreditCardsAddressesStorage(
+    context: Context
+) : CreditCardsAddressesStorage, AutoCloseable {
+
+    private val coroutineContext by lazy { Dispatchers.IO }
```
[the remark this answers is not in the document]
Comment (grigoryk) [reply components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/AutofillCreditCardsAddressesStorage.kt:31]: Well, I can't find anything relevant in Sentry. Will poke around more later on, but for now I reckon this is fine.

File: components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/AutofillCreditCardsAddressesStorage.kt [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+    private val conn by lazy {
+        AutofillStorageConnection.init(dbPath = context.getDatabasePath(AUTOFILL_DB_NAME).absolutePath)
+        AutofillStorageConnection
+    }
+
+    init {
+        // Set the name of the native library so that we use
+        // the appservices megazord for compiled code.
```
Comment (grigoryk) [inline components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/AutofillCreditCardsAddressesStorage.kt:40]: Link to https://github.com/mozilla/application-services/issues/3874 to not forget to remove this.

**Review by grigoryk — APPROVED** [review]
👍 Looks good to me, let's land it and iterate in tree!

**Pull request merged**